server/src/main/java/org/elasticsearch/ingest/OnFailureProcessor.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.ingest;
+
+import org.elasticsearch.core.Tuple;
+
+import java.util.List;
+
+/**
+ * A wrapping processor that adds failure handling logic around the wrapped processor.
+ */
+public class OnFailureProcessor extends CompoundProcessor implements WrappingProcessor {
+
+    static final String TYPE = "on_failure";
+
+    public OnFailureProcessor(boolean ignoreFailure, Processor processor, List<Processor> onFailureProcessors) {
```

Comment:
I don't think we have any test coverage (in OnFailureProcessorTests or IngestStatsNamesAndTypesIT) for when `onFailureProcessors` is not empty, do we? Might also be worth documenting that that is non-empty if there is an `on_failure` block, and empty if `ignore_failure` = true.